using-mobile-extensions/adobe-campaignclassic/adobe-campaignclass-c-api-reference.md

```diff
@@ -0,0 +1,261 @@
+					@Override
+					public void call(final Boolean status) {
+						Log.w(TAG, "Status: " + status;
+					}
+				});
+  	}
+}
+```
+
+{% endtab %}
+
+{% tab title="iOS" %}
+
+Follow [Apple's instructions](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/HandlingRemoteNotifications.html#//apple_ref/doc/uid/TP40008194-CH6-SW1) to get your app ready to handle push notifications. Once you receive the APNS token, you will need to send that token along with the device information to Campaign Classic using the registerDevice API.
+
+### registerDevice
+
+The registerDevice API will register a device with your Campaign Classic registration server. It takes the APNS token as a parameter along with a user key that identifies the user such as an email address or login name. It can also be given a dictionary of custom key value pairs that you want to associate with the registration. Lastly, a boolean value is returned in the callback which signals whether the registration was sucessful.
+
+#### Objective-C
+
+#### Syntax
+
+```objectivec
++ (void) registerDevice: (nonnull NSData*) token userKey: (nonnull NSString*) userKey additionalParams: (nullable NSDictionary*) additionalParams callback: (nullable void (^) (BOOL success)) callback;
+```
+
+#### Example
+
+```objectivec
+- (void) application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
+  // Set the deviceToken that the APNS has assigned to the device
+  NSMutableDictionary *params = [[NSMutableDictionary alloc] initWithObjectsAndKeys:      	@"Adobe", @"brand", nil];
+  [params setObject: [NSNumber numberWithInt:12345] forKey: @"testNum"];
+  [params setObject: [NSNumber numberWithBool:YES]  forKey: @"testBool"];
+    
+[ACPCampaignClassic registerDevice:deviceToken userKey:@"user@gmail.com" additionalParams:params callback:^(BOOL success) {
+	NSLog(@"Status: %d", success);
+}
+```
+
+#### Swift
+
+```swift
+ACPCampaignClassic.registerDevice(deviceToken, userKey: userKey, additionalParams: additionalParams, callback: {(_ success: Bool?) -> Void in
+            NSLog("Status: %d", success)                                                                                                                        })
+```
+
+{% endtab %}
+{% endtabs %}
+
+## trackNotification API
+
+### Tracking for push message receive and clickthrough
+
+Adobe Campaign Classic has two additional API used for tracking push messages received and push messages clicked.
+
+### trackNotificationReceive
+
+The trackNotificationReceive API should be called after a push message has been received in the app. On iOS it takes a dictionary as a parameter and the dictionary should contain the message id and delivery id retrieved from the received push message. On Android, the API takes a map containing the same message id and delivery id received from the push message.
```

Comment:
- Please add a comma between "parameter" and "and", so that it reads as follows:
"...a dictionary as a parameter, and the dictionary should..."

- Please capitalize "id" to "ID"